file.go

```diff
@@ -293,15 +293,25 @@ func (f *File) reload(s dataSource) error {
 
 // Reload reloads and parses all data sources.
 func (f *File) Reload() (err error) {
-	for _, s := range f.dataSources {
+	for i, s := range f.dataSources {
 		if err = f.reload(s); err != nil {
+			// Notice: if ShortCircuit and Loose are both true, ShortCircuit has priority over Loose.
+			// Return err when all configuration sources fail to load.
+			if f.options.ShortCircuit {
+				if i < len(f.dataSources)-1 {
+					continue
+				}
```

Comment:
Could you elaborate why this check is required?

Reply:
When all configuration sources fail to be loaded, the error information about the last configuration source is returned.
In other words, no need to `continue` when loading the last configuration source.

Reply:
>the error information about the last configuration source is returned.

Why is that a problem?

Reply:
I mean, shouldn't we return error when there is an error?

Reply:
In my opinion, in Short Circuit mode, when one of multiple configuration sources can be successfully loaded, nil should be returned. When all configuration sources fail to be loaded, the last error message should be returned.

Is there any deviation in my understanding?

Reply:
I think ShortCircuit means that it returns when there is one successfully file loaded, it is not equivalent to swallow errors.

However, to make it really useful in practice, we need to combine the behavior of Loose, i.e. ignore non-existent files.

So I think ShortCircuit and Loose are not conflict to each other but complementary. Does this make sense to you?

Reply:
I see. I agree with you. 
I'll update the PR later.